.github/workflows/tests.yaml

```
@@ -0,0 +1,47 @@
+name: Tests
+
+on: [push, workflow_dispatch]
+
+jobs:
+  get-go-version:
+    name: "Determine Go toolchain version"
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.go-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Determine Go version
+        id: get-go-version
+        run: |
+          echo "Building with Go $(cat .go-version)"
+          echo "::set-output name=go-version::$(cat .go-version)"
```

To avoid `set-output` deprecation warnings, most of these actions should be bumped up a version (details: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

```suggestion
      - uses: actions/checkout@v3
      - name: Determine Go version
        id: get-go-version
        run: |
          echo "Building with Go $(cat .go-version)"
          echo "go-version=$(cat .go-version)" >> $GITHUB_OUTPUT
```